test/metabase/lib/card_test.cljc

```diff
@@ -211,3 +212,23 @@
                        (concat (from :source/implicitly-joinable (cols-of :people)))
                        sorted)
                   (->> nested lib.metadata.calculation/visible-columns sorted))))))))
+
+(deftest ^:parallel display-name-of-joined-cards-is-clean-test
+  (testing "We get proper field names rather than ids (#27323)"
+    (let [query (lib/query lib.tu/metadata-provider-with-mock-cards (:products lib.tu/mock-cards))
+          people-card (:people lib.tu/mock-cards)
+          lhs (m/find-first (comp #{"ID"} :name) (lib/join-condition-lhs-columns query 0 people-card nil nil))
+          rhs (m/find-first (comp #{"ID"} :name) (lib/join-condition-rhs-columns query 0 people-card nil nil))
+          join-clause (lib/join-clause people-card [(lib/= lhs rhs)])
+          query (lib/join query join-clause)
+          filter-col (m/find-first (comp #{"Mock people card__ID"} :lib/desired-column-alias)
+                                   (lib/filterable-columns query))
+          query (-> query
+                    (lib/filter (lib/= filter-col 1))
+                    (lib/aggregate (lib/distinct filter-col))
+                    (as-> $q (lib/breakout $q (m/find-first (comp #{"SOURCE"} :name)
+                                                            (lib/breakoutable-columns $q)))))]
+      (is (=
+           [["Source" "Distinct values of ID"] ["ID is equal to 1"]]
+           [(map #(lib/display-name query %) (lib/returned-columns query))
+            (map #(lib/display-name query %) (lib/filters query))])))))
```

Comment:
<i class='graphite__hidden'>[Re: lines 231 to 235]<br/></i>

I think this reads a lot better as two `is` clauses. The column and filter checks aren't really related.<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/metabase/metabase/35515?utm_source=unchanged-line-comment">Graphite</a>.</span>